src/core/CLucene/index/DocumentsWriterThreadState.cpp

```diff
@@ -582,6 +583,38 @@ bool DocumentsWriter::ThreadState::postingEquals(const TCHAR* tokenText, const i
   int32_t pos = p->textStart & CHAR_BLOCK_MASK;
 
   int32_t tokenPos = 0;
+
+#ifdef _AVX2
+  constexpr auto AVX2_SIZE = sizeof(__m256i);
+  constexpr auto tcharSize = sizeof(TCHAR);
+  size_t tcharBytes = tokenTextLen * tcharSize;
+  int loop = 0;
+  if (tcharBytes >= AVX2_SIZE) {
+    const __m256i* p1 = reinterpret_cast<const __m256i*>(tokenText + tokenPos);
+    const __m256i* p2 = reinterpret_cast<const __m256i*>(text + pos);
+    __m256i all_ones = _mm256_set1_epi8(0xFF);
+    loop = tcharBytes / AVX2_SIZE;
+    for (int i = 0; i < loop; ++i) {
+        if (i < loop - 2) {
+          _mm_prefetch(reinterpret_cast<const char*>(p1) + AVX2_SIZE * 2, _MM_HINT_T0);
+          _mm_prefetch(reinterpret_cast<const char*>(p2) + AVX2_SIZE * 2, _MM_HINT_T0);
+        }
+
+        __m256i v1 = _mm256_loadu_si256(p1++);
+        __m256i v2 = _mm256_loadu_si256(p2++);
+        __m256i cmp = _mm256_cmpeq_epi8(v1, v2);
+
+        if (!_mm256_testc_si256(cmp, all_ones)) { // Not all bytes are equal
+            return false;
+        }
+    }
+  }
+  if (loop != 0) {
+    tokenPos += (loop * AVX2_SIZE / tcharSize);
+    pos += (loop * AVX2_SIZE / tcharSize);
+  }
```

Comment:
this can be moved into the previous `if`

Reply:
done